components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt

```diff
@@ -57,6 +64,14 @@ class SitePermissionsFeature(
 
     private val observer = SitePermissionsRequestObserver(sessionManager, feature = this)
 
+    internal val storage by lazy { storageInitializer() }
+    internal val ioCoroutineScope by lazy { CoroutineScope(Dispatchers.IO) }
+
+    @VisibleForTesting
+    internal var storageInitializer = {
+        SitePermissionsStorage(anchorView.context)
+    }
+
```

Comment:
I think we should pass this in as a parameter to the constructor of the feature. The reason for this is that the app probably will need this storage too (since it may build app-specific UI). And I assume we want to avoid that there are multiple instances of this?

Reply:
Sound reasonable 👍